src/main.rs

```diff
@@ -0,0 +1,254 @@
+use std::path::PathBuf;
+
+use clap::Parser;
+
+const DEFAULT_COMPRESSION_LEVEL: usize = 6;
```

Comment:
Why usize, rather than something like int?

Reply:
I had anticipated that whatever zlib bindings we chose would use `usize`. I think we've settled on [flate2](https://github.com/rust-lang/flate2-rs), though, so this should be a `u32`.